src/digest.rs

```diff
@@ -0,0 +1,86 @@
+use crate::raw;
+use std::os::raw::c_char;
+
+/// `Digest` is a high-level rust interface to the Valkey module C API
+/// abstracting away the raw C ffi calls.
+pub struct Digest {
+    pub dig: *mut raw::RedisModuleDigest,
+}
+
+impl Digest {
+    pub const fn new(dig: *mut raw::RedisModuleDigest) -> Self {
+        Self { dig }
+    }
+
+    /// Returns the key name of this [`Digest`].
+    ///
+    /// # Panics
+    ///
+    /// Will panic if `RedisModule_GetKeyNameFromDigest` is missing in redismodule.h
+    pub fn get_key_name(&self) -> *const raw::RedisModuleString {
```

Comment:
To help the caller functions not operate on raw pointers, can we update this to return a `ValkeyString` type?

Reply:
Yes, updated